File: src/main/java/com/github/zxh/classpy/classfile/constant/ConstantStringInfo.java
package com.github.zxh.classpy.classfile.constant;

/*
CONSTANT_String_info {
    u1 tag;
    u2 string_index;
}
*/
public class ConstantStringInfo extends ConstantInfo {

    {
        u2("string_index");
    }

    @Override
    protected String loadDesc(ConstantPool cp) {
        int stringIndex = super.getUInt("string_index");
        return cp.getUtf8Info(stringIndex).loadDesc(cp);
    }
    
}

File: src/main/kotlin/com/github/zxh/classpy/classfile/constant/ConstantStringInfo.kt
package com.github.zxh.classpy.classfile.constant;

/*
CONSTANT_String_info {
    u1 tag;
    u2 string_index;
}
*/
class ConstantStringInfo : ConstantInfo() {

    init {
        u2("string_index");
    }

    override fun loadDesc(cp: ConstantPool): String {
        val stringIndex = super.getInt("string_index");
        return cp.getUtf8Info(stringIndex).loadDesc(cp);
    }
    
}
